Source/MediaInfo/Multiple/File_MpegTs.cpp

```diff
@@ -1922,7 +1922,8 @@ void File_MpegTs::Read_Buffer_Unsynched()
         {
             #ifdef MEDIAINFO_MPEGTS_PESTIMESTAMP_YES
                 Complete_Stream->Streams[StreamID]->Searching_ParserTimeStamp_Start_Set(false); //No more searching start
-                if (((File_MpegPs*)Complete_Stream->Streams[StreamID]->Parser)->HasTimeStamps)
+                if (Complete_Stream->Streams[pid]->Parser->ParserName == "MpegPs"

```

Comment:
I am not fan on testing the parser name, I would prefer a different fix e.g. initialize HasTimeStamps in all cases.

Reply:
In File__Analyze class ?